sql/core/src/main/scala/org/apache/spark/sql/execution/BufferedRowIterator.java

```diff
@@ -61,6 +63,14 @@ public long durationMs() {
   public abstract void init(Iterator<InternalRow> iters[]);
 
   /**
+   * Initializes from array of iterators of InternalRow.
+   */
+  public void init(int index, Iterator<InternalRow> iters[]) {
+    partitionIndex = index;
```

Comment:
The problem is the case of two RDDs. Under this case, we will use the original API.